--- a/node.gyp
+++ b/node.gyp
@@ -720,6 +720,7 @@
           'libraries': [
             'Dbghelp',
             'Psapi',
+            'winmm',
             'Ws2_32',
           ],
         }],
--- a/node.gyp
+++ b/node.gyp
@@ -255,6 +255,14 @@
             'winmm.lib',
             'Ws2_32.lib',
           ],
+          'conditions': [
+            [ 'node_shared=="true"', { 
+              'dependencies!': [
+                'tools/v8_gypfiles/v8.gyp:v8_snapshot',
+                'tools/v8_gypfiles/v8.gyp:v8_libplatform',
+              ],
+            }],
+          ],
         }],
         ['node_with_ltcg=="true"', {
           'msvs_settings': {
--- a/node.gypi
+++ b/node.gypi
@@ -29,7 +29,7 @@
     [ 'clang==1', {
       'cflags': [ '-Werror=undefined-inline', ]
     }],
-    [ 'node_shared=="false" and "<(_type)"=="executable"', {
+    [ '"<(_type)"=="executable"', {
       'msvs_settings': {
         'VCManifestTool': {
           'EmbedManifest': 'true',
--- a/tools/install.py
+++ b/tools/install.py
@@ -123,6 +123,7 @@ def corepack_files(action):
 def subdir_files(path, dest, action):
   ret = {}
   for dirpath, dirnames, filenames in os.walk(path):
+    dirpath = dirpath.replace("\\","/")
     files_in_path = [dirpath + '/' + f for f in filenames if f.endswith('.h')]
     ret[dest + dirpath.replace(path, '')] = files_in_path
   for subdir, files_in_path in ret.items():